### PR TITLE
fix(#2370): Better "y/N" prompts

### LINKED ISSUE #2370: Better [Y/n] Prompts
Packer uses prompts like: `OK to remove? [y/N]`

It's a bit more robust and friendly than our current solution of prepopulating the default.

### PR DESCRIPTION
**Pull request by mohamedarish**
fixes #2370.

Packer uses a bool flag comparing to check whether the given input is "y".

I've made it so that the default behaviour of the prompt is set to N for no (don't delete).

The default is set to R(ename) for name conflicts when copy-pasting.

Accidentally clicking enter/return after pressing d/bd should be made less common with this default behaviour.

the default behaviour is shown as a capital letter.

### CODE REVIEW

**Comment by mohamedarish** [comment]
I've thought about it and the "no" condition is unnecessary as everything that is not "y" is a no condition. I'm closing this PR and would be reopening this after I've fixed this issue.

**Comment by alex-courtis** [comment]
Thanks for taking the time. This one is indeed more difficult than it appears.

**Comment by mohamedarish** [comment]
I've removed all "No" conditions as they are not needed as everything that is not a "y" or "yes" entered is a no condition. And would delete/replace the files

**Review by alex-courtis — CHANGES_REQUESTED** [review]
Many thanks for your contribution. This greatly enhances UX.

- [x] maybe address default `r`
- [x] Please update `lib.prompt` to convert the callback's item_short to lowercase. Users may start using uppercase to match the prompt.

File: lua/nvim-tree/actions/fs/copy-paste.lua [hunk 1]
```diff
@@ -110,8 +110,8 @@ local function do_single_paste(source, dest, action_type, action_fn)
       end)
     else
       local prompt_select = "Overwrite " .. dest .. " ?"
-      local prompt_input = prompt_select .. " y/n/r(ename): "
-      lib.prompt(prompt_input, prompt_select, { "y", "n", "r" }, { "Yes", "No", "Rename" }, function(item_short)
+      local prompt_input = prompt_select .. " R(ename)/y/n: "
+      lib.prompt(prompt_input, prompt_select, { "r", "y" }, { "Rename", "Yes" }, function(item_short)
```
[the remark this answers is not in the document]
Comment (mohamedarish) [reply lua/nvim-tree/actions/fs/copy-paste.lua:114]: anything that does not start with y or n shall become r in that case?
This feels wrong and having don't copy be the default case feels better

[the remark this answers is not in the document]
Comment (mohamedarish) [reply lua/nvim-tree/actions/fs/copy-paste.lua:114]: I have already started working on making it lowercase

[the remark this answers is not in the document]
Comment (alex-courtis) [reply lua/nvim-tree/actions/fs/copy-paste.lua:114]: > anything that does not start with y or n shall become r in that case?
> This feels wrong and having don't copy be the default case feels better

Yes please. Default `n` is consistent and arguably better behaviour. All prompts default to no.

It's a somewhat breaking change however it is quite reasonable.

**Comment by mohamedarish** [comment]
I've noticed that my change makes the prompt ui from [dressing.nvim](https://github.com/stevearc/dressing.nvim) or [telescope-ui-select.nvim
](https://github.com/nvim-telescope/telescope-ui-select.nvim) and have updated the items_long to fix this and also fixed to change all item_short to be lowercase.

File: lua/nvim-tree/actions/fs/copy-paste.lua (hunk 1 above)
[the remark this answers is not in the document]
Comment (mohamedarish) [reply lua/nvim-tree/actions/fs/copy-paste.lua]: Currently "r" is auto filled when a name conflict on copy occurs and I feel that is a better behaviour than defaulting to don't copy in that case.
The user needs to remove the r or update the autofilled character as it provides a second chance to cancel by cancelling the rename prompt

**Comment by alex-courtis** [comment]
> I've noticed that my change makes the prompt ui from [dressing.nvim](https://github.com/stevearc/dressing.nvim) or [telescope-ui-select.nvim ](https://github.com/nvim-telescope/telescope-ui-select.nvim) not have proper select prompts (Default(No) appears as is and does not fit into the prompt ui) and have updated the items_long to fix this and also fixed to change all item_short to be lowercase.

Thank you for testing.

File: lua/nvim-tree/actions/fs/copy-paste.lua (hunk 1 above)
[the remark this answers is not in the document]
Comment (alex-courtis) [reply lua/nvim-tree/actions/fs/copy-paste.lua]: > Currently "r" is auto filled when a name conflict on copy occurs and I feel that is a better behaviour than defaulting to don't copy in that case.

I'm happy with `r` as the default, however auto-filling is not consistent with other prompts. We can remove the auto-fill and rename can still be the default.

**Review by alex-courtis — CHANGES_REQUESTED** [review]
Please:

- [x] resolve exception
- [x] remove auto-fill r

File: lua/nvim-tree/actions/fs/copy-paste.lua [hunk 2]
```diff
@@ -110,8 +110,8 @@ local function do_single_paste(source, dest, action_type, action_fn)
       end)
     else
       local prompt_select = "Overwrite " .. dest .. " ?"
-      local prompt_input = prompt_select .. " y/n/r(ename): "
-      lib.prompt(prompt_input, prompt_select, { "y", "n", "r" }, { "Yes", "No", "Rename" }, function(item_short)
+      local prompt_input = prompt_select .. " R(ename)/y/n: "
+      lib.prompt(prompt_input, prompt_select, { "r", "y", "n" }, { "Rename", "Yes", "No" }, function(item_short)
```
[the remark this answers is not in the document]
Comment (mohamedarish) [reply lua/nvim-tree/actions/fs/copy-paste.lua:114]: I have already resolved the exception, Will remove the auto-fill r

[the remark this answers is not in the document]
Comment (mohamedarish) [reply lua/nvim-tree/actions/fs/copy-paste.lua]: I have fixed the exception and added another condition in the elseif block for rename for rename to be "r"/"R" or ""

**Review by alex-courtis — APPROVED** [review]
All cases tested OK.

Many thanks for your contribution.

**Pull request merged**